tests/aws/services/stepfunctions/legacy/test_stepfunctions_legacy.py

```diff
@@ -721,6 +723,7 @@ def test_default_logging_configuration(create_state_machine, custom_client):
         assert result["ResponseMetadata"]["HTTPStatusCode"] == 200
         assert result["loggingConfiguration"] == {"level": "OFF", "includeExecutionData": False}
     finally:
+        custom_client.stepfunctions.delete_state_machine(stateMachineArn=result["stateMachineArn"])
```

Comment:
The test did not use `custom_client` to create the statemachine and then failed during retrieval using this client.

Affected test: test_default_logging_configuration